### PR TITLE
Fix username & email uniqueness check for batch PATCHes

### PR DESCRIPTION
**Pull request by thgreasi**
Change-type: patch
Signed-off-by: Thodoris Greasidis <thodoris@balena.io>

### CODE REVIEW

File: src/features/auth/hooks/validate-username-email.ts [hunk 1]
```diff
@@ -8,26 +8,44 @@ const USERNAME_BLACKLIST = ['root'];
...
-				if (existingUser) {
-					throw new ConflictError('This email is already taken');
-				}
 			}
 		},
 	});
 }
+
+for (const method of ['POST', 'PATCH'] as const) {
+	hooks.addPureHook(method, 'resin', 'user', {
```
Comment (Page-) [inline src/features/auth/hooks/validate-username-email.ts:23]:
```suggestion
```

File: src/features/auth/hooks/validate-username-email.ts [hunk 2]
```diff
@@ -8,26 +8,42 @@ const USERNAME_BLACKLIST = ['root'];
...
-			if (request.values.email != null) {
-				const existingUser = await findUser(request.values.email, tx, ['id']);
-				if (existingUser) {
-					throw new ConflictError('This email is already taken');
-				}
 			}
 		},
 	});
+
+	hooks.addPureHook(method, 'resin', 'user', {
```
Comment (Page-) [inline src/features/auth/hooks/validate-username-email.ts:21]:
```suggestion
	});

	hooks.addPureHook(method, 'resin', 'user', {
```

Comment (thgreasi) [reply src/features/auth/hooks/validate-username-email.ts]: I guess you mean to combine the two right? B/c the code suggestion doesn't seem to have any change :thinking:

File: src/features/auth/hooks/validate-username-email.ts [hunk 3]
```diff
@@ -8,26 +8,39 @@ const USERNAME_BLACKLIST = ['root'];
...
+					if (request.values[field] == null) {
+						return;
+					}
+					if (method === 'PATCH') {
+						const affectedIds = await sbvrUtils.getAffectedIds(args);
+						if (affectedIds.length === 0) {
+							return;
+						}
+						if (affectedIds.length > 1) {
+							throw new ConflictError(`The ${field} must be unique`);
```
Comment (Page-) [inline src/features/auth/hooks/validate-username-email.ts:32]: Can you put a `// TODO: This should be handled by a rule for case insensitive uniqueness` please

**Review by Page- — APPROVED** [review]

**Pull request merged**